README.md

```diff
@@ -68,7 +71,310 @@ You can combine `consumer_id` and `service_id` in the same request, to furthermo
+request upstream. Now try sending request with header `x-user-id` set
+
+```bash
+$ curl -i -X GET localhost:8000/requests/user/foo \
+  -H "X-User-Id:bob"
+```
+
+This time plugin will add a new header `x-consumer-id` with value sent along
+with header `x-user-id`, i.e.`bob`
+
+## Order of execution
+
+Plugin performs the response transformation in the following order:
+
+* remove → rename → replace → add → append
+
+## Examples
+
+<div class="alert alert-info.blue" role="alert">
+  <strong>Kubernetes users:</strong> version <code>v1beta1</code> of the Ingress
+  specification does not allow the use of named regex capture groups in paths.
+  If you use the ingress controller, you should use unnamed groups, e.g.
+  <code>(\w+)/</code> instead of <code>(?&lt;user_id&gt;\w+)</code>. You can access
+  these based on their order in the URL path, e.g. <code>$(uri_captures[1])</code>
+  will obtain the value of the first capture group.
+</div>
+
+In these examples we have the plugin enabled on a Service. This would work
+similarly for Routes.
+
+- Add multiple headers by passing each header:value pair separately:
+
+{% tabs %}
+{% tab With a database %}
+```bash
+$ curl -X POST http://localhost:8001/services/example-service/plugins \
+  --data "name=request-transformer" \
+  --data "config.add.headers[1]=h1:v1" \
+  --data "config.add.headers[2]=h2:v1"
+```
+{% tab Without a database %}
+```yaml
+plugins:
+- name: request-transformer
+  config:
+    add:
+      headers: ["h1:v1", "h2:v1"]
+```
+{% endtabs %}
+
+<table>
+  <tr>
+    <th>incoming request headers</th>
+    <th>upstream proxied headers:</th>
+  </tr>
+  <tr>
+    <td>h1: v1</td>
+    <td>
+      <ul>
+        <li>h1: v1</li>
+        <li>h2: v1</li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+- Add multiple headers by passing comma separated header:value pair (only possible with a database):
+
+```bash
+$ curl -X POST http://localhost:8001/services/example-service/plugins \
+  --data "name=request-transformer" \
+  --data "config.add.headers=h1:v1,h2:v2"
+```
+
+<table>
+  <tr>
+    <th>incoming request headers</th>
+    <th>upstream proxied headers:</th>
+  </tr>
+  <tr>
+    <td>h1: v1</td>
+    <td>
+      <ul>
+        <li>h1: v1</li>
+        <li>h2: v1</li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+- Add multiple headers passing config as JSON body (only possible with a database):
+
+```bash
+$ curl -X POST http://localhost:8001/services/example-service/plugins \
+  --header 'content-type: application/json' \
+  --data '{"name": "request-transformer", "config": {"add": {"headers": ["h1:v2", "h2:v1"]}}}'
+```
+
+<table>
+  <tr>
+    <th>incoming request headers</th>
+    <th>upstream proxied headers:</th>
+  </tr>
+  <tr>
+    <td>h1: v1</td>
+    <td>
+      <ul>
+        <li>h1: v1</li>
+        <li>h2: v1</li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+- Add a querystring and a header:
+
+{% tabs %}
+{% tab With a database %}
```

Comment:
Should we change those to plain Markdown instead? I don't believe GitHub understands our templating directives.